<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AllWheelForm</class>
 <widget class="QWidget" name="AllWheelForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>801</width>
    <height>545</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>All-Wheel Steering</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_9">
   <property name="margin">
    <number>10</number>
   </property>
   <item>
    <widget class="QWidget" name="upperVerticalLayout" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="margin">
       <number>0</number>
      </property>
      <item>
       <widget class="QWidget" name="explicitVerticalLayout" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>300</width>
          <height>300</height>
         </size>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="explicitTitleLabel">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="font">
            <font>
             <pointsize>14</pointsize>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>Explicit Wheel Control:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="explicitWidget" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>300</width>
             <height>325</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>300</width>
             <height>365</height>
            </size>
           </property>
           <widget class="QLabel" name="sketchLabel">
            <property name="geometry">
             <rect>
              <x>20</x>
              <y>10</y>
              <width>241</width>
              <height>271</height>
             </rect>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="pixmap">
             <pixmap resource="LunaGUIResources.qrc">:/images/sketch.png</pixmap>
            </property>
            <property name="scaledContents">
             <bool>true</bool>
            </property>
           </widget>
           <widget class="QLineEdit" name="rfDrivingEdit">
            <property name="geometry">
             <rect>
              <x>220</x>
              <y>90</y>
              <width>31</width>
              <height>22</height>
             </rect>
            </property>
           </widget>
           <widget class="QPushButton" name="sendButton">
            <property name="geometry">
             <rect>
              <x>143</x>
              <y>280</y>
              <width>121</width>
              <height>32</height>
             </rect>
            </property>
            <property name="text">
             <string>Send</string>
            </property>
           </widget>
           <widget class="QLineEdit" name="lrDrivingEdit">
            <property name="geometry">
             <rect>
              <x>40</x>
              <y>180</y>
              <width>31</width>
              <height>22</height>
             </rect>
            </property>
           </widget>
           <widget class="QLineEdit" name="rfSteeringEdit">
            <property name="geometry">
             <rect>
              <x>150</x>
              <y>90</y>
              <width>31</width>
              <height>22</height>
             </rect>
            </property>
           </widget>
           <widget class="QLineEdit" name="lrSteeringEdit">
            <property name="geometry">
             <rect>
              <x>100</x>
              <y>180</y>
              <width>31</width>
              <height>22</height>
             </rect>
            </property>
           </widget>
           <widget class="QLineEdit" name="rrSteeringEdit">
            <property name="geometry">
             <rect>
              <x>150</x>
              <y>180</y>
              <width>31</width>
              <height>22</height>
             </rect>
            </property>
           </widget>
           <widget class="QPushButton" name="resetButton">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>280</y>
              <width>131</width>
              <height>32</height>
             </rect>
            </property>
            <property name="text">
             <string>Reset</string>
            </property>
           </widget>
           <widget class="QLineEdit" name="lfSteeringEdit">
            <property name="geometry">
             <rect>
              <x>100</x>
              <y>90</y>
              <width>31</width>
              <height>22</height>
             </rect>
            </property>
           </widget>
           <widget class="QLineEdit" name="rrDrivingEdit">
            <property name="geometry">
             <rect>
              <x>220</x>
              <y>180</y>
              <width>31</width>
              <height>22</height>
             </rect>
            </property>
           </widget>
           <widget class="QLineEdit" name="lfDrivingEdit">
            <property name="geometry">
             <rect>
              <x>30</x>
              <y>90</y>
              <width>31</width>
              <height>22</height>
             </rect>
            </property>
           </widget>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="predefinedAndICRWidget" native="true">
        <property name="maximumSize">
         <size>
          <width>153</width>
          <height>16777215</height>
         </size>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <widget class="QWidget" name="ICRWidget" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>160</width>
             <height>0</height>
            </size>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <property name="margin">
             <number>0</number>
            </property>
            <item>
             <widget class="QWidget" name="predefinedActionsWidget" native="true">
              <layout class="QVBoxLayout" name="verticalLayout">
               <property name="margin">
                <number>0</number>
               </property>
               <item>
                <widget class="QLabel" name="predefinedTitleLabel">
                 <property name="font">
                  <font>
                   <pointsize>14</pointsize>
                   <weight>75</weight>
                   <bold>true</bold>
                  </font>
                 </property>
                 <property name="text">
                  <string>Predefined actions:</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="forwardButton">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>Drive Forward</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="backwardButton">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>Drive Back</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="leftButton">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>Crab Left</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="rightButton">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>Crab Right</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="turnLeftButton">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>Turn CCW</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="turnRightButton">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>Turn CW</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="stopButton">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>Stop</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="ICRControlTitleLabel">
              <property name="font">
               <font>
                <pointsize>14</pointsize>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>ICR Control</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QWidget" name="ICRFormWidget" native="true">
              <layout class="QFormLayout" name="formLayout">
               <property name="leftMargin">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>5</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <item row="0" column="0">
                <widget class="QLabel" name="ICRXTitleLabel">
                 <property name="text">
                  <string>x:</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QLineEdit" name="ICRXEdit">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="minimumSize">
                  <size>
                   <width>30</width>
                   <height>0</height>
                  </size>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="ICRYTitleLabel">
                 <property name="text">
                  <string>y:</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QLineEdit" name="ICRYEdit">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="minimumSize">
                  <size>
                   <width>30</width>
                   <height>0</height>
                  </size>
                 </property>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="ICRVelTitleLabel">
                 <property name="text">
                  <string>velocity:</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QLineEdit" name="velocityEdit">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="minimumSize">
                  <size>
                   <width>30</width>
                   <height>0</height>
                  </size>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="sendICRButton">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="maximumSize">
               <size>
                <width>150</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>Send ICR</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGraphicsView" name="graphicsView">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="styleSheet">
         <string notr="true">background: transparent</string>
        </property>
        <property name="frameShape">
         <enum>QFrame::NoFrame</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Plain</enum>
        </property>
        <property name="interactive">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="stateWidget" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>123</height>
      </size>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_8">
      <property name="margin">
       <number>0</number>
      </property>
      <item>
       <widget class="QLabel" name="stateTitleLabel">
        <property name="font">
         <font>
          <pointsize>14</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Platfotm state:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="stateHorizontalLayout" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <widget class="QWidget" name="drivingVerticalLayout" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>0</height>
            </size>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <property name="margin">
             <number>0</number>
            </property>
            <item>
             <widget class="QLabel" name="drivingTitleLabel">
              <property name="text">
               <string>Driving:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QWidget" name="drivingGridLayout" native="true">
              <layout class="QGridLayout" name="gridLayout_2">
               <property name="margin">
                <number>0</number>
               </property>
               <item row="0" column="0">
                <widget class="QLabel" name="dlfLabel">
                 <property name="text">
                  <string>? Rad/s</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QLabel" name="drfLabel">
                 <property name="text">
                  <string>? Rad/s</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="dlrLabel">
                 <property name="text">
                  <string>? Rad/s</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QLabel" name="drrLabel">
                 <property name="text">
                  <string>? Rad/s</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="steeringVerticalLayout" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>0</height>
            </size>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <property name="margin">
             <number>0</number>
            </property>
            <item>
             <widget class="QLabel" name="steeringTitleLabel">
              <property name="text">
               <string>Steering:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QWidget" name="steeringGridLayout" native="true">
              <layout class="QGridLayout" name="gridLayout">
               <property name="margin">
                <number>0</number>
               </property>
               <item row="0" column="0">
                <widget class="QLabel" name="slfLabel">
                 <property name="text">
                  <string>? Rad</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QLabel" name="srfLabel">
                 <property name="text">
                  <string>? Rad</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="slrLabel">
                 <property name="text">
                  <string>? Rad</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QLabel" name="srrLabel">
                 <property name="text">
                  <string>? Rad</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="ICRVerticalLayout" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_7">
            <property name="margin">
             <number>0</number>
            </property>
            <item>
             <widget class="QLabel" name="ICRTitleLabel">
              <property name="text">
               <string>ICR:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="ICRLabel">
              <property name="text">
               <string>?, ?</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="LunaGUIResources.qrc"/>
 </resources>
 <connections/>
</ui>
